vendor/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+
+if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+# Update submodules as needed
+    option(GIT_SUBMODULE "Check submodules during build" ON)
+    if(GIT_SUBMODULE)
+        message(STATUS "Submodule update")
+        execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
+                        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                        RESULT_VARIABLE GIT_SUBMOD_RESULT)
+        if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+            message(FATAL_ERROR "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+        endif()
+    endif()
+endif()
```

Comment:
This should probably be replaced with the [newer approach used in CSE](https://github.com/cse-sim/cse/blob/86e65c39c3b2cb4adb9f199aadf02f55c5eb18f2/vendor/CMakeLists.txt#L1-L2).

Reply:
Happy to update it, but just for my clarification, does initialize_submodules just catch the case when submodule directory doesn't yet exist? Which could likely happen for new installs?

Reply:
Yeah, it just makes sure that the submodules are cloned if they haven't been already.